src/main/webapp/one-off/databaseInit.java

```diff
@@ -40,104 +39,123 @@
 
-        recipeEntity.setProperty("title", title);
-        recipeEntity.setProperty("index_title", title.toLowerCase());
-        recipeEntity.setProperty("imgURL", imgURL);
-        recipeEntity.setProperty("ingredients", ingredients);
-        recipeEntity.setProperty("stepList", stepList);
-        recipeEntity.setProperty("author", "Piece of Cake");
-        recipeEntity.setProperty("difficulty", "N/A");
-        recipeEntity.setProperty("prep_time", "N/A");
-        recipeEntity.setProperty("cook_time", "N/A");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    KeyRange keyRange = datastore.allocateIds("Recipe", 1L);
 
-        datastore.put(recipeEntity);
-    }
+    Entity recipeEntity = new Entity(keyRange.getStart());
+    recipeEntity.setProperty("title", title);
+    recipeEntity.setProperty("index_title", title.toLowerCase());
+    recipeEntity.setProperty("imgURL", imgURL);
+    recipeEntity.setProperty("ingredients", ingredients);
+    recipeEntity.setProperty("stepList", stepList);
+    recipeEntity.setProperty("author", "Piece of Cake");
 
-    /**
-      * readFileInList reads a text file in a list of strings, each string being a line in the original file
-     **/
-    public static List<String> readFileInList(String fileName) throws Exception, IOException  { 
-  
-         List<String> lines = Collections.emptyList(); 
-         lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8); 
-         return lines; 
-    } 
+    Document recipeDocument = Document
+      .newBuilder()
+      .setId(String.valueOf(recipeEntity.getKey().getId()))
+      .addField(Field.newBuilder().setName("title").setText(title))
+      .addField(Field.newBuilder().setName("imgURL").setText(imgURL))
+      .build();
 
-    /**
-      * parseRecipe calls 'readFileInList' on a given text file, extrapolates all the recipes properties from the list of lines, and inputes them in 'upload'
-     **/
-    public void parseRecipe(String fileName) throws Exception, IOException  {
+    datastore.put(recipeEntity);
+    index.put(recipeDocument);
+  }
 
-       List<String> lines = readFileInList("INSERT ABSOLUTE PATH HERE"+fileName);
-       String title = lines.get(2);
-       String imgURL = lines.get(4);
-       int currentLineIndex = 7;
+  public static List<String> readFileInList(String fileName)
+    throws Exception, IOException {
+    List<String> lines = Collections.emptyList();
+    lines = Files.readAllLines(Paths.get(fileName), StandardCharsets.UTF_8);
+    return lines;
+  }
 
-       ArrayList<String> ingredients = new ArrayList<String>();
-       while (!lines.get(currentLineIndex).isBlank()){
-            ingredients.add(lines.get(currentLineIndex));
-            currentLineIndex += 1;
-       }
+  public void parseRecipe(String fileName) throws Exception, IOException {
+    List<String> lines = readFileInList(
+      "/home/donpaul/step/capstone/step249-2020/src/main/webapp/one-off/recipes/" +
```

Comment:
This should be defined in a constant outside the main class, or as a constant of the class.

RECIPES_DIRECTORY